tests/ut/ir/transforms/test_flatten_tile_nd_to_2d.py

```
@@ -2401,5 +2401,116 @@ def main(self, x: pl.Tensor[[1, 16, 128], pl.FP16]) -> pl.Tensor[[1, 16, 128], p
+                x_tile: pl.Tile[[2, 3, 4], pl.FP32] = pl.tile.load(x, [0, 0, 0], [2, 3, 4])
+                xt_tile: pl.Tile[[3, 2, 4], pl.FP32] = pl.transpose(x_tile, axis1=0, axis2=1)
+                out_0: pl.Tensor[[3, 2, 4], pl.FP32] = pl.tile.store(xt_tile, [0, 0, 0], out_0)
+                return out_0
+
+            @pl.function
+            def main(self, x: pl.Tensor[[2, 3, 4], pl.FP32]) -> pl.Tensor[[3, 2, 4], pl.FP32]:
+                out_0: pl.Tensor[[3, 2, 4], pl.FP32] = pl.create_tensor([3, 2, 4], dtype=pl.FP32)
+                y: pl.Tensor[[3, 2, 4], pl.FP32] = self.main_incore_0(x, out_0)
+                return y
+
+        with pytest.raises(ValueError, match="only last-two-axes tile.transpose"):
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)